pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -136,6 +136,12 @@ type ActualStateOfWorld interface {
 	// have no mountedPods. This list can be used to determine which volumes are
 	// no longer referenced and may be globally unmounted and detached.
 	GetUnmountedVolumes() []AttachedVolume
+
+	// GetActualPodVolumeSpecName generates and returns a map of pods in which map
+	// is indexed with pod uid and value is a map of the pod's mounted volume spec
+	// names. This map can be used to determine which pod and volume is currently
+	// in actual state of world based on pod uid and volume spec name.
+	GetActualPodVolumeSpecNames() map[string]map[string]bool
```

Comment:
We can probably simplify this and have it just return true/false if pod exists. If pod exists then we assume that the volumes exist. As long as the check is on Pod UID. Or if you prefer, return a map of pod UIDs that you can check against. Regardless, we don't need a full list of volumes for the pod.